docs/api/baseplate/observers/timeout.rst

```diff
@@ -0,0 +1,62 @@
+Server Timeouts
+===============
+
+The timeout observer ends processing of requests in your service if they take
+too long. This is particularly important when an upstream service times out on
+its end and retries requests to your services which will cause a pileup.
+
+This is entirely configured in-service at the moment and no headers from
+upstream services are yet taken into account.
+
+.. warning::
+
+   The timeout mechanism is entirely cooperative. If request processing is
+   taking a long time because it is doing compute-heavy actions and not
+   yielding to the event loop it might go on longer than the allotted timeout.
+
+Configuration
+-------------
+
+Make sure your service calls
+:py:meth:`~baseplate.Baseplate.configure_observers` during application startup.
+By default, requests will time out after 10 seconds. The following
+configuration settings allow you to customize this.
+
+.. code-block:: ini
+
+   [app:main]
+
+   ...
+
+   # optional: defaults to 10 seconds if not specified. this timeout
+   # is used for any endpoint not specified in the by_endpoint
+   # section below.
+   server_timeout.default = 200 milliseconds
+
+   # optional: defaults to false. if enabled, tracebacks will be
+   # printed to the logs when timeouts occur.
+   server_timeout.debug = true
+
+   # optional: timeout values for specific endpoints. the name
+   # used must match the name of the server span generated.
+   # this overrides the default timeout.
+   # - thrift services: the name of the thrift RPC method
+   # - pyramid services: the name of the route
```

Comment:
🔕  It might be useful to call attention to where the name of the route is defined to make it easier for devs to know where to look for it.

https://docs.pylonsproject.org/projects/pyramid/en/1.10-branch/api/config.html#pyramid.config.Configurator.add_route